### PR TITLE
Generation 4: Fix Transform's interaction with items

### PR DESCRIPTION
**Pull request by ascriptmaster**
Causes Dittos transforming into Giratina-O without Griseous Orb to revert
to Giratina-A in Generation 4, etc.

### CODE REVIEW

**Comment by ascriptmaster** [comment]
I might be missing something in Gen4, so I'll call @Marty-D again

**Comment by Marty-D** [comment]
Looks good so far! The only other pre-Gen 5 Transform difference I can think of right now is that it actually copies species. So for example, a Ditto holding Thick Club Transfomed into a Marowak gets the boost. Species stopped being copied in Gen 5, so the opposite happens now; Marowak holding Thick Club Transformed into anything else gets the boost.

**Comment by ascriptmaster** [comment]
So, does that mean that in Generation 4 you can't Knock Off a Griseous Orb on a Ditto transformed into Giratina, or a Plate on a Ditto transformed into Arceus?

**Comment by ascriptmaster** [comment]
So the species item check thing would be something like this?

Also, because it is relevant for the Griseous Orb, if I create `mods/gen4/rulesets.js` and add a `validateSet` to `BattleFormats.pokemon` in there, will it override the original validator function?

File: mods/gen4/abilities.js [hunk 1]
```diff
@@ -64,6 +64,15 @@ exports.BattleAbilities = {
 		rating: 0,
 		num: 58
 	},
+	"multitype": {
+		inherit: true,
+		onTakeItem: function (item, pokemon, source) {
```
Comment (Marty-D) [inline mods/gen4/abilities.js]: Mold Breaker will ignore this!

Comment (ascriptmaster) [inline mods/gen4/abilities.js]: Should I just hardcode it then?

**Comment by ascriptmaster** [comment]
I hardcoded the Arceus check for Gen 4 and added the validation for non-Giratina Griseous Orb.

Don't merge quite yet, I typed this up really fast on the train and haven't had the chance to test whether or not it works.

File: mods/gen4/items.js [hunk 2]
```diff
@@ -106,6 +138,30 @@ exports.BattleItems = {
 		gen: 4,
 		desc: "Holder's damaging moves do 1.3x damage; loses 1/10 max HP after the attack."
 	},
+	"lightball": {
+		inherit: true,
+		onModifySpA: function (spa, pokemon) {
+			if (pokemon.template.baseSpecies === 'Pikachu') {
```
Comment (TheImmortal) [inline mods/gen4/items.js]: Could have put species as Pikachu doesn't have formes in gen 4.

Comment (ascriptmaster) [inline mods/gen4/items.js]: Done.

**Comment by ascriptmaster** [comment]
I tested my Arceus code and the Griseous Orb code, so once Marty code reviews one more time it's ready for merge.

**Comment by Marty-D** [comment]
One last thing I missed when I looked at it earlier - it's Multitype by itself that prevents items being given or taken in Gen 4, not Arceus with Multitype (although it's only relevant in a hackmons setting). Everything else looks good to me, nicely done!

**Comment by ascriptmaster** [comment]
Bulbapedia said it only works on Arceus, but I guess they meant the type changing part and not the item retention part. I'll fix once I get home

**Comment by ascriptmaster** [comment]
Multitype item removal check now only checks for Multitype.

**Comment by ascriptmaster** [comment]
Still waiting for @Marty-D's opinion

**Comment by Marty-D** [comment]
:+1: (for some reason I thought I merged this yesterday)

**Pull request merged**